### PR TITLE
Use java classes for reflection in MiskWebFormBuilder

### PR DESCRIPTION
**Pull request by tso**
The differences in reflection on kotlin vs java classes
was causing kotlin compiled protos to fail to be usable
on the admin dashboard for services. This aligns the behavior
of the two.

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt [hunk 1]
```diff
@@ -48,15 +50,24 @@ class MiskWebFormBuilder {
...
+        for (property in klass.primaryConstructor?.parameters ?: listOf()) {
+          val fieldName = property.name ?: "UNKNOWN"
+          if (fieldName != "unknownFields") {
+            handleField(property.type, fieldName, fields, stack)
+          }
+        }
+      } else {
+        for (property in klass.declaredMemberProperties) {
+          // Use the WireField annotation to identify fields of our proto.
+          if (property.annotations.any { it is WireField }) {
```
Comment (tso) [inline misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt:63]: The kotlin generated protos don't return any annotations

File: misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt [hunk 2]
```diff
@@ -48,15 +50,24 @@ class MiskWebFormBuilder {
 
       val fields = mutableListOf<Field>()
 
-      for (property in klass.declaredMemberProperties) {
-        // Use the WireField annotation to identify fields of our proto.
-        if (property.annotations.any { it is WireField }) {
-          val fieldName = property.name
-          handleField(property.returnType, fieldName, fields, stack)
+      if (klass.companionObject != null) {
```
Comment (tso) [inline misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt:53]: Hacky way of determining kotlin vs java proto

File: misk/src/test/kotlin/misk/web/metadata/AddCreditDecisionRequest.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
+// Code generated by Wire protocol buffer compiler, do not edit.
```
Comment (tso) [inline misk/src/test/kotlin/misk/web/metadata/AddCreditDecisionRequest.kt:1]: Copy and pasted, need to fix gradle files to generate this in misk?

File: misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt [hunk 4]
```diff
@@ -34,60 +35,64 @@ class MiskWebFormBuilder {
...
+    val stack = LinkedList<Class<*>>()
+    stack.push(requestClass.java)
 
     while (stack.isNotEmpty()) {
       val klass = stack.pop()
 
       // No need to re-process a given type.
       // This acts as the visited set of our type graph traversal.
-      if (typesMap.containsKey(klass.qualifiedName!!)) {
+      if (typesMap.containsKey(klass.canonicalName!!)) {
```
Comment (adrw) [inline misk/src/main/kotlin/misk/web/MiskWebFormBuilder.kt:46]: Is this still a `klass` or is it now a `class` (Java)?

File: misk/src/test/kotlin/misk/web/metadata/MiskWebFormBuilderTest.kt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      Field(
+        name = "status",
+        type = "Enum<com.squareup.protos.test.parsing.Shipment.State,VALIDATING,PICKING_UP," +
+          "DELIVERING,CONSUMING>",
+        repeated = false
+      )
+    )
+  }
+
+  @Test fun `handles kotlin wire messages`() {
```
Comment (adrw) [inline misk/src/test/kotlin/misk/web/metadata/MiskWebFormBuilderTest.kt:60]: beautiful

File: misk/src/test/kotlin/misk/web/metadata/protos/Robot.kt [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+import kotlin.Any
+import kotlin.Boolean
+import kotlin.Int
+import kotlin.Long
+import kotlin.String
+import kotlin.hashCode
+import kotlin.jvm.JvmField
+import okio.ByteString
+
+public class Robot(
```
Comment (adrw) [inline misk/src/test/kotlin/misk/web/metadata/protos/Robot.kt:21]: Do these classes need to be checked in or could they live in protos folder and be generated when tests are run? Seems like they could grow stale and lose effectiveness if Wire changes the generated code etc.

**Review by adrw — APPROVED** [review]
Great find and fix! Many thanks!

File: misk/src/test/kotlin/misk/web/metadata/protos/Robot.kt (hunk 6 above)
Comment (tso) [reply misk/src/test/kotlin/misk/web/metadata/protos/Robot.kt:21]: Yeah definitely agree :) I was hoping to have them compiled but when I paired with @swankjesse he said this was probably best

**Review by swankjesse — APPROVED** [review]

**Pull request merged**